metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/utils/EntityMergeEngine.java

```diff
@@ -409,19 +410,20 @@ public void mergeEntities(RDF rdf, List<EnrichmentBaseWrapper> enrichmentBaseWra
    * @param proxyLinkTypes Lookup of the link types to create in the europeana proxy. The keys are
    * the about values of the entities to add.
    */
-  public void mergeEntities(RDF rdf, List<EnrichmentBase> contextualEntities,
+  public void mergeEntities(RDF rdf, List<List<EnrichmentBase>> contextualEntities,
       Map<String, Set<EnrichmentFields>> proxyLinkTypes) {
-    for (EnrichmentBase entity : contextualEntities) {
-      final Set<String> links = getSameAsLinks(entity);
-      links.add(entity.getAbout());
+    for (List<EnrichmentBase> entitiesList : contextualEntities) {
+      final Set<String> links = entitiesList.stream().map(this::getSameAsLinks).flatMap(Set::stream)
+          .collect(Collectors.toSet());
+      entitiesList.forEach(entity -> links.add(entity.getAbout()));
       final Set<EnrichmentFields> fields = links.stream().map(proxyLinkTypes::get)
           .filter(Objects::nonNull).flatMap(Set::stream).collect(Collectors.toSet());
-      convertAndAddEntity(rdf, entity, fields);
+      entitiesList.forEach(entity -> convertAndAddEntity(rdf, entity, fields));
     }
   }
 
 
-  private static Set<String> getSameAsLinks(EnrichmentBase contextualClass) {
+  private Set<String> getSameAsLinks(EnrichmentBase contextualClass) {
```

Comment:
Why not leave this static?

Reply:
Good question, I don't remember why I removed the static, will add it again